ccm_web/server/src/api/dtos/api.section.users.dto.ts

```diff
@@ -0,0 +1,33 @@
+import { ArrayMaxSize, ArrayMinSize, IsArray, IsIn, IsNotEmpty, ValidateNested } from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+import { Type } from 'class-transformer'
+
+export class SectionUserDto {
```

Comment:
Do you think it would it be clearer/more accurate to call this a `SectionEnrollment`? As I scan the PR, I see the variable shorthand often ends up as `user(s)`, which is somewhat confusing given the preexisting presence of `user`. And really, once we add an enrollment type for a context, we've moved past it being a simple Canvas user object. This would require some tweaking through, but might help readability.

Reply:
So, rather than the enrollment code referring to "user(s)", refer to "enrollment(s)"?  I think the problem with that is I was thinking of the input as a "user" because Canvas' API accepts a "user" object and returns an object called "enrollment".

I'm happy to make changes that clarify what the code is doing, but I think that change would make it less clear.

Reply:
I think we could probably debate this for a while. You are creating an enrollment, so you give it enrollment data (not just data about a user -- you really only provide a user identifier -- but also a role in that context, when that enrollment begins and ends, etc.). The endpoint describes the object provided in the body as an enrollment (https://canvas.instructure.com/doc/api/enrollments.html#method.enrollments_api.create).That's consistent with creating a course or a section: you give it some course and section data, it gives you the course or section it created back.

So, while I don't quite follow your logic, if it's going to make it less clear to you, you can keep it as is. It's not terribly confusing to me now, so it's not worth getting into a back-and-forth over. I'll leave it to your discretion.